### PR TITLE
Nidx query

### PR DESCRIPTION
**Pull request by javitonino**
Query planner & friends for NIDX

### CODE REVIEW

**Review by lferran — APPROVED** [review]

File: nidx/nidx_types/src/lib.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@
 //
 use std::{collections::HashSet, path::PathBuf};
 
+pub mod query_language;
```
[the remark this answers is not in the document]
Comment (javitonino) [reply nidx/nidx_types/src/lib.rs:22]: I don't think it is. In the `query_language` mod I only copied the types. The actual logic to build those types and such is in the `nidx` crate. In `nidx_types` we just have the types needed to express boolean expressions and such.

I think the name is still fair.

File: nidx/nidx_vector/src/lib.rs [hunk 2]
```diff
@@ -72,11 +74,8 @@ impl VectorIndexer {
             .collect();
 
         // Do the merge
-        let open_destination = data_point::merge(
-            work_dir,
-            &segment_ids.iter().map(|(a, b)| (a, b)).collect::<Vec<_>>(),
-            &VectorConfig::default(),
-        )?;
+        let open_destination =
+            data_point::merge(work_dir, &segment_ids.iter().map(|(a, b)| (a, b)).collect::<Vec<_>>(), &config)?;
```
[the remark this answers is not in the document]
Comment (javitonino) [reply nidx/nidx_vector/src/lib.rs:78]: More or less. It's taking references to the elements in the vector (from `Vec<(a,b)>` to `Vec<(&a,&b)>`).

File: nidx/tests/common.rs [hunk 3]
```diff
@@ -18,9 +18,6 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 //
 
-// rust doesn't detect usage of `common` in our integration tests
-#![allow(dead_code)]
-
```
[the remark this answers is not in the document]
Comment (javitonino) [reply nidx/tests/common.rs:23]: Oh, didn't know that. I'll revert this.

File: nidx/nidx_vector/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jotare) [reply nidx/nidx_vector/src/lib.rs:78]: Sometimes is hard to read some stuff without inlay type hints :)

File: nidx/tests/common.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jotare) [reply nidx/tests/common.rs]: Yeah, not the nicest feature :)

**Pull request merged**